src/client/app/components/visual-unit/CreateVisualUnitModalMapComponent.tsx

```diff
@@ -0,0 +1,219 @@
+			/* auto: point towards dest. node */
+			.attr('orient', 'auto')
+			.append('svg:path')
+			.attr('d', 'M0,-5L10,0L0,5');
+
+		/* Start arrow head (for bidirectional edges) */
+		svg.append('defs').append('marker')
+			.attr('id', 'arrow-start')
+			.attr('viewBox', '0 -5 10 10')
+			.attr('refX', 25)
+			.attr('refY', 0)
+			.attr('markerWidth', 4)
+			.attr('markerHeight', 4)
+			/* auto-start-reverse: point towards src. node */
+			.attr('orient', 'auto-start-reverse')
+			.append('svg:path')
+			.attr('d', 'M0,-5L10,0L0,5');
+
+		/* Link style */
+		const link = svg.selectAll('line')
+			.data(links)
+			.enter().append('line')
+			.style('stroke', '#aaa')
+			.attr('stroke-width', 3)
+			.attr('marker-end', 'url(#arrow-end)')
+			/* Only draw start arrow head if bidirectional */
+			.attr('marker-start', d => d.bidirectional === true ? 'url(#arrow-start)' : '');
+
+		/* Node style */
+		const node = svg.selectAll('.node')
+			.data(nodes)
+			.enter().append('circle')
+			/* Node radius */
+			.attr('r', 20)
+			/* checks if unit has a non empty suffix to color differently */
+			.attr('fill', d => d.suffix && d.typeOfUnit === 'unit' ? colorSchema('suffix.input') : colorSchema(d.typeOfUnit));
+
+		/* Drag behavior */
+		node.call(d3.drag()
+			.on('start', dragstart)
+			.on('drag', dragged)
+			.on('end', dragend));
+
+		/* Node label style */
+		const label = svg.selectAll('.label')
+			.data(nodes)
+			.enter()
+			.append('text')
+			.text(function (d) { return d.name; })
+			.style('text-anchor', 'middle')
+			.style('fill', '#000')
+			.style('font-family', 'Arial')
+			.style('font-size', 14);
+
+		/* Update element positions when moved */
+		simulation.on('tick', () => {
+			link
+				.attr('x1', d => d.source.x)
+				.attr('y1', d => d.source.y)
+				.attr('x2', d => d.target.x)
+				.attr('y2', d => d.target.y);
+
+			node
+				.attr('cx', d => d.x)
+				.attr('cy', d => d.y);
+
+			label
+				.attr('x', function(d){ return d.x; })
+				.attr('y', function (d) {return d.y - 25; });
+		});
+
+		// eslint-disable-next-line jsdoc/require-jsdoc
+		function dragstart(event: any) {
+			if (!event.active) simulation.alphaTarget(0.3).restart();
+			event.subject.fx = event.subject.x;
+			event.subject.fy = event.subject.y;
+		}
+
+		// eslint-disable-next-line jsdoc/require-jsdoc
+		function dragged(event: any) {
+			event.subject.fx = event.x;
+			event.subject.fy = event.y;
+		}
+
+		// eslint-disable-next-line jsdoc/require-jsdoc
+		function dragend(event: any) {
+			if (!event.active) simulation.alphaTarget(0);
+			event.subject.fx = null;
+			event.subject.fy = null;
+		}
+
+		/* Color Legend */
+		const legend = svg.append('g')
+			.attr('transform', `translate(${-width / 2 + 20}, ${-height / 2 + 20})`);
+
+		colorSchema.domain().forEach((item, i) => {
+			const legendEntry = legend.append('g')
+				.attr('transform', `translate(0, ${i * 25})`);
+
+			// Rectangle color box
+			legendEntry.append('rect')
```

Comment:
This is a very small point but is it easy to make this a circle instead of a rectangle so the legendf matches the item in the graph? (other visual too)